### PR TITLE
Linting changes

### PR DESCRIPTION
**Pull request by vladsaveliev**
* Use conda in the GitHub workflow to install a dev environment containing pre-commit and pylint.
  Using https://pre-commit.com/#conda for pre-commit wasn't a good option after all. This feature is a way to ask pre-commit to install the linters using conda instead of cloning the repos. However, pre-commit itself still needed to be installed first, so there is doesn't seem to be away to get away with only one environment file.
* Run `pre-commit run` explicitly in the GitHub workflow instead of running the linters stand-alone.

```
    - name: pre-commit
      run: |
        conda activate test-env
        pre-commit run --all-files
```

  (There is also a [GitHub action](https://github.com/pre-commit/action) for that, but there is no easy way to activate the conda environment for it, and it doesn't add value.)

* In order to enable inspections of python module imports with pylint, I switched to a `local` setup of pylint in pre-commit config:

```
- repo: local
  hooks:
  - id: pylint
    name: pylint
    entry: pylint
    language: system
    types: [python]
```

I didn't set up the same for `flake8`, `black` and `pre-commit-hooks` as it would require specifying the `name`,` entry` and `language` value for every hook id, and would make `.pre-commit-config.yaml` unreadable. I think it's fine that `pre-commit` creates its own hidden virtual environment, as normally you don't need access to individual linters explicitly: one can run all of them in the command line with one command `pre-commit run` (also documented now).
* Moving checking of single quotes from a pylint plugin into a flake8 plugin.
* Adding more documentation on disabling the inspections.
* Excluding `.txt` and `.tsv` files for the trailing-whitespace hook.
* Disable the `fixme` inspection (that checks for TODO).
* Increasing the max number of local variables from 15 to 25 and the max number of function args from 5 to 10.
* Adding markdownlint into pre-commit (cc @pdiakumis - let me know if you have comments!). Renaming the config file to `.markdownlint.json` (seems to be more standard?) and excluding a couple more inspections.
* Adding linters and the GitHub workflow into the `team-docs` repo itself, and run the Pulumi python code through the python linters.

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  lint:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: conda-incubator/setup-miniconda@v2
```
Comment (lgruen) [inline .github/workflows/lint.yaml:14]: You can use `with` here to specify the file and which environment to activate:
https://github.com/populationgenomics/analysis-runner/pull/5/files#diff-71cabc4177e41ea8f15a89eb65398fa8187739f68a2762706d84885cd8b2ddc3R17

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    - name: Install packages
+      run: |
+        conda env create --name test-env -f environment-dev.yml
```
Comment (lgruen) [inline .github/workflows/lint.yaml:18]: Slight inconsistency here between the environment called `test` vs `dev`.

**Review by lgruen — APPROVED** [review]
Using the local pylint is a great solution! Thanks for figuring that one.

Added a minor comment, therefore not merging directly.

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (vladsaveliev) [reply .github/workflows/lint.yaml:14]: The issue with that is that you can't specify the environment name like with `--name`, and you have to use the same name as in the `environment.yaml` file. And I'd love the workflow to be transferrable between projects with copy-and-paste without needing to modify.

File: .github/workflows/lint.yaml (hunk 2 above)
Comment (vladsaveliev) [reply .github/workflows/lint.yaml:18]: Same: `environment-dev.yml` will have a project-specific name; and I wanted the GitHub workflow to be generic enough to be able to copy it around, that's why a generic environment name.

**Pull request merged**